app/services/partners/request_create_service.rb

```diff
@@ -54,16 +54,27 @@ def populate_item_request(partner_request)
         attrs['item_id'].blank? && attrs['quantity'].blank?
       end
 
-      item_requests = formatted_line_items.map do |ira|
-        Partners::ItemRequest.new(
-          item_id: ira['item_id'],
-          quantity: ira['quantity'],
-          children: ira['children'] || [], # will create ChildItemRequests if there are any
-          name: fetch_organization_item_name(ira['item_id']),
-          partner_key: fetch_organization_partner_key(ira['item_id'])
-        )
+      items = {}
+
+      formatted_line_items.each do |input_item|
+        pre_existing_entry = items[input_item['item_id']]
+        if pre_existing_entry
+          pre_existing_entry.quantity = (pre_existing_entry.quantity.to_i + input_item['quantity'].to_i).to_s
+          # TODO: Is the following the correct way to merge?
+          pre_existing_entry.children += input_item['children'] || []
+        else
+          items[input_item['item_id']] = Partners::ItemRequest.new(
+            item_id: input_item['item_id'],
+            quantity: input_item['quantity'],
+            children: input_item['children'] || [], # will create ChildItemRequests if there are any
+            name: fetch_organization_item_name(input_item['item_id']),
+            partner_key: fetch_organization_partner_key(input_item['item_id'])
```

Comment:
We're relying on the fact that a single request doesn't come from more than one partner (which is why we're not doing any merging of name / partner).

Reply:
yeah. Probably all of this needs additional refactoring well beyond this de-duplication.